src/child/actions/sidebar.js

```diff
@@ -16,3 +17,27 @@ export const selectFavourites = createActionCreator(() => ({
         action: 'Favourites'
     }
 }));
+
+const windowReopened = createActionCreator((reopenWindowName) => ({
+    type: ACTION_TYPES.REOPEN_WINDOW,
+    reopenWindowName
+}));
+
+const openWindowError = createActionCreator((reopenWindowName) => ({
+    type: ACTION_TYPES.OPEN_WINDOW_ERROR,
+    reopenWindowName
+}));
+
+export function reopenWindow(windowName) {
+    return dispatch => {
+        const childWindow = new fin.desktop.Window(
+            configService.getWindowConfig(windowName),
+            () => childWindow.show(
+                false,
+                () => dispatch(windowReopened(windowName)),
+                () => dispatch(openWindowError())
```

Comment:
If a message isn't used to reopen the window, as suggested above, then I think this should be passed the `windowName`?